web/packages/shared/components/ClusterDropdown/ClusterDropdown.tsx

```diff
@@ -160,20 +180,76 @@ export function ClusterDropdown({
         open={Boolean(anchorEl)}
         onClose={handleClose}
       >
-        {options.map(cluster => (
-          <MenuItem
-            px={2}
-            key={cluster.value}
-            onClick={() => onChangeOption(cluster.value)}
-          >
-            <Text ml={2} fontWeight={300} fontSize={2}>
-              {cluster.label}
-            </Text>
-          </MenuItem>
-        ))}
+        <Flex flexDirection="column">
+          {/* the cluster filter should show on total number of options rather than filtered options to ensure the input isn't lost when filtering */}
+          {options.length > 5 && (
+            <ClusterFilterWrapper>
+              <ClusterFilter
+                p="2"
+                m="2"
+                type="text"
+                autoFocus
+                value={clusterFilter}
+                autoComplete="off"
+                onChange={onClusterFilterChange}
+                placeholder={'Search clusters…'}
+              />
+            </ClusterFilterWrapper>
+          )}
+          <Box css={clusterOptionsCss}>
+            {options
+              .filter(cluster =>
+                cluster.label
+                  .toLowerCase()
+                  .includes(clusterFilter.toLowerCase())
+              )
+              .map(cluster => (
+                <MenuItem
+                  px={2}
+                  key={cluster.value}
+                  onClick={() => onChangeOption(cluster.value)}
+                >
+                  <Text
+                    ml={2}
+                    fontWeight={cluster.value === clusterId ? 500 : 300}
+                    fontSize={2}
+                  >
+                    {cluster.label}
+                  </Text>
+                </MenuItem>
+              ))}
+          </Box>
+        </Flex>
       </Menu>
     </Flex>
   );
 }
 
 type Option = { value: string; label: string };
+
+const menuCss = () => `
+  top: 240px !important;
```

Comment:
matches the gap with the other filter dropdown
```suggestion
  top: 236px !important;
```